adjustments for mobile devices

amieszczak opened the pull request:
I added some responsive design for the site. Dimensions and distances between elements are still to be set properly.
I have some questions:
-  am I allowed to use block-type element inside 'flex'?
- is the name of class containing 'side' instead of 'site' proper here?
- in mobile devices, organization of elements inside 'middle-page-content' div switch from horizontal to vertical one. In case of self-describing names of classes - 'left-side-content' and 'right-side-content' loses its meaning. How can I omit the problem? Maybe I can make new classes with 'display: none' as default and turn them on for mobile devices while turning off 'side' type classes?

amieszczak committed:
adjustments for mobile devices

schmitshny reviewed (changes requested):
![Screenshot 2023-11-26 at 15 17 30](https://github.com/amieszczak/sample-page-design/assets/33054342/532e36c0-3a04-4223-a895-f54f8766d696)
![Screenshot 2023-11-26 at 15 17 43](https://github.com/amieszczak/sample-page-design/assets/33054342/662231fc-2ddc-4f7c-82be-58d495fed3bd)

amieszczak committed:
responsive design

amieszczak committed:
adjustments for larger/smaller mobile devices

amieszczak committed:
- button elements hover states;
- menu - logo elements proximity;
- organization of features for dropdown;
- dropdown animation;
- 2 dropdown dropdown animations with toggle behaviour;
- sliding menu for mobile devices;
- opening and closing of elements behaviour;
- mobile-menu slide menu;
- dropdown in slide menu;
